python/lsst/sims/catalogs/db/dbConnection.py

```diff
@@ -370,10 +370,7 @@ def get_column_names(self, tableName=None):
         """
         tableNameList = self.get_table_names()
         if tableName is not None:
-            if tableName not in tableNameList:
-                return []
-            else:
-                return [str_cast(xx['name']) for xx in reflection.Inspector.from_engine(self.connection.engine).get_columns(tableName)]
+            return [str_cast(xx['name']) for xx in reflection.Inspector.from_engine(self.connection.engine).get_columns(tableName)]
```

Comment:
If you just want to expand the tableNameList to include 'views' (i.e. SummaryAllProps), you could modify this class when it generates the tableNameList like so: 

self.tableNames = reflection.Inspector.from_engine(self.connection.engine).get_table_names()
self.tableNames += reflection.Inspector.from_engine(self.connection.engine).get_view_names()

Not sure if this is a better option or not - your alternative is fine (but if you had a strong motivation for making an all-inclusive tableNameList, this other option would solve that). 

I don't think you need the str_cast, do you? (maybe you do in your applications? I've found that just 
self.columnNames[t] = [xxx['name'] for xxx in cols]
 seems to work for the stuff in MAF).

Reply:
The `str_cast` is something I had to do during the great python2->3 switch of 2016.  It seems harmless enough, I won't poke at it.

You're probably right that I should just add `get_view_names()` to the list of valid "tables."  That is the least offensive change to behavior.